src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java

```diff
@@ -18,16 +19,20 @@
   /**
    * text and confidence (and any additional meta data) to be published
    */
-  public static class RecognizedResult {
+  public static class ListeningEvent {
+    public long ts;
     public Double confidence;
+    public boolean isSpeaking = false;
+    public boolean isListening = false;
+    public boolean isRecording = false;
     public boolean isFinal = false;
     public String text;
   }
 
   /**
    * generalized list of languages and their codes - if useful
    */
-  static protected Map<String, String> languages = new HashMap<>();
+  protected Map<String, Locale> locales = new HashMap<>();

```

Comment:
No Required if this implements the LocaleInterface.  I don't think you need/want this local variable here.  The interface should be enfoced.  if you want the locals. you should use the setters/getters from the interface..  not this local protected member.  this should be removed.

Reply:
interface is enforced, this is "set" in the constructor by the interfaces getLocales() which the derived class must implement.  Its a convenience property, because otherwise the webgui would need to subscribe to getLocales explicitly. In this case, all services sent to javascript contain their appropriate locales.